pinot-core/src/main/java/org/apache/pinot/core/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -142,6 +148,18 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
       } else {
         _invertedIndex = null;
       }
+
+
+      if (loadFSTIndex) {
+        if (segmentReader.hasIndexFor(columnName, ColumnIndexType.FST_INDEX)) {
```

Comment:
Like the way it is done for other indexes, I don't think we check here if the index is existing or not. 
The creation of the respective index reader will fail and the segment load will fail to let the user know that IndexLoadingConfig was incorrectly built from IndexingConfig

Reply:
fixed